source/opt/module.cpp

```diff
@@ -172,6 +172,21 @@ uint32_t Module::GetExtInstImportId(const char* extstr) {
   return 0;
 }
 
+void Module::EnsureIdDefinedBeforeInstruction(uint32_t id, inst_iterator inst) {
```

Comment:
If you make the other changes, this function should be unnecessary.

Reply:
Done